components/security/src/lib.rs

```diff
@@ -19,20 +19,28 @@ use grpcio::{
     ServerCredentialsFetcher,
 };
 use log_wrappers::RedactOption;
+use online_config::{ConfigChange, ConfigManager, OnlineConfig, Result as CfgResult};
+use tikv_util::info;
 
-#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Default)]
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Default, OnlineConfig)]
 #[serde(default)]
 #[serde(rename_all = "kebab-case")]
 pub struct SecurityConfig {
     // SSL configs.
+    #[online_config(skip)]
     pub ca_path: String,
+    #[online_config(skip)]
     pub cert_path: String,
+    #[online_config(skip)]
     pub key_path: String,
     // Test purpose only.
     #[serde(skip)]
+    #[online_config(skip)]
     pub override_ssl_target: String,
+    #[online_config(skip)]
     pub cert_allowed_cn: HashSet<String>,
-    pub redact_info_log: RedactOption,
+    pub redact_info_log: String,
```

Comment:
legit unit test error:

```
thread 'config::tests::test_compatibility_with_old_config_template' panicked at src/config/mod.rs:7220:70:

called `Result::unwrap()` on an `Err` value: Error { inner: ErrorInner { kind: Custom, line: Some(1233), col: 19, at: Some(51640), message: "invalid type: boolean `false`, expected a string", key: ["security", "redact-info-log"] } }
```

This change failed the `test_compatibility_with_old_config_template` test --- existing config which used `redact_info_log = false` will no longer work after this PR.